src/black/nodes.py

```diff
@@ -789,6 +789,16 @@ def is_import(leaf: Leaf) -> bool:
     )
 
 
+def is_with_stmt(leaf: Leaf) -> bool:
+    """Return True if the given leaf starts a with statement."""
+    return bool(
+        leaf.type == token.NAME
+        and leaf.value == "with"
```

Comment:
Does this also cover `async with`? We should treat both of these the same.

Reply:
Good catch! Turns out `async with` isn't handled at all in #3489. It has a different code path [here](https://github.com/psf/black/blob/25d886f52c2bbbb58386ac8050f4e67952507bc7/src/black/linegen.py#L289) so the code I'm touching here isn't called at all.

I filed #3591, will address separately.